documentation/Spring-native-vs-Quarkus.asciidoc

```diff
@@ -0,0 +1,104 @@
+
+*Note: Spring Native is still in Beta. The current version of Spring
+Native 0.10.5 is designed to be used with Spring Boot 2.5.6.*
+
+Spring Native provides support for compiling Spring applications to
+native executables using the GraalVM native-image compiler. Using a
+native image provides some key advantages, such as instant startup,
+instant peak performance, and reduced memory consumption. But there are
+also some drawbacks: Creating a native image is a heavy process which is
+slower than a regular application. A native image also has fewer runtime
+optimizations after its warmup. Furthermore, it is less mature than the
+JVM with some different behaviors.
+
+*Difference in regular JVM vs GraalVM:*      
+
+* Static analysis of the application from the main entry point is
+performed at build time.
+* Unused parts are removed at build time.
+* Configuration required for reflection, resources, and dynamic proxies.
+* Classpath is fixed at build time.
+* No class lazy loading: everything shipped in the executables will be
+loaded in memory on startup.
+* Some code will run at build time.
+
+There are https://github.com/oracle/graal/blob/master/docs/reference-manual/native-image/Limitations.md[limitations] around some aspects of Java applications that are not fully supported and therefore should be aware of before using Spring Native.
+
+=== Build time and start time for apps
+
+[cols=",,",options="header",]
+|===
+|Framework |build time |start time
+|Spring Native |19.615s |2.913s
+|Quarkus |52.818s |0.802s
+|===
+
+=== Memory footprints
+
+[cols=",",options="header",]
+|===
+|Framework |memory footprint
+|Spring Native |109 MB
+|Quarkus |75 MB
+|===
+
+=== Considering devonfw best practices:
+
+devonfw actively supports Spring as it is widely used and is unmatched regarding stability and support, even though being slower in build and start times and having a bigger memory footprint in general. 
+While Spring Native solves these problems quite well, it is still in beta and therefore not considered stable enough to be an option for devonfw as of now. 
+devonfw Spring modules could be supported in a Spring Native application but there is no guarantee that it will work.
+Despite being quite new, we still consider Quarkus as stable and mature enough to be supported eventhough there are no Quarkus modules in devonfw at the moment.
+
+=== General recommendations and conclusion:
+
+For Greenfield projects which are mainly focusing on microservices or the cloud, Quarkus can be recommended as it is more suitable for their requirements. 
+Spring projects, should be able to use Spring Native in the future in order to improve their performance. 
+Despite devonfw not officially supporting Spring Native, you still can read https://docs.spring.io/spring-native/docs/current/reference/htmlsingle/[how to migrate from Spring to Spring Native] if you want to try things out. 
+We also provide a link:getting-started-for-spring-developers.asciidoc[guide
```

Comment:
This link does not work because the referenced guide is located in the quarkus subfolder
-->  link:quarkus/getting-started-for-spring-developers.asciidoc